src/Particle/SoaDistanceTableAA.h

```diff
@@ -128,19 +125,28 @@ struct SoaDistanceTableAA : public DTD_BConds<T, D, SC>, public DistanceTableDat
           min_dist = temp_r_[jat];
           index    = jat;
         }
-      if (index >= 0)
-        dr = temp_dr_[index];
+      assert(index >= 0);
```

Comment:
This sort of makes up for the lack of unit test, but I'd rather there was one and it was run in debug. At the moment I can't see how anything but temp_r_'s of all NaN or infty could trip this.  What insures the calling code never produces such input?

Reply:
I don't think we can protect NaN anywhere in the code. I never tound this causing an issue. So I just leave the assert.

Reply:
I think given the limited use of this function we can table it, at least there is an assert.